### PR TITLE
Home page search improvements

### PR DESCRIPTION
**Pull request by OliverRadini**
still not really styled, but you can now press enter to search, and you can now click a player's name to visit their profile

### CODE REVIEW

**Comment by andir** [comment]
LGTM 👍 
(besides the minor whitespace issues)

**Review by burfo — CHANGES_REQUESTED** [review]
Nice work!  Please fix the lint errors.

P.S. you can run the lint `npm run lint` and the tests `npm run test` beforehand to see these errors.

File: herald/src/app/shared/data-table/data-table.component.ts [hunk 1]
```diff
@@ -16,18 +16,31 @@ export class DataTableComponent implements OnInit {
...
             this.updateData(data);
         });
     }
 
+    getHeaderLink(headerKey): string{
+        let foundWord = this.linkColumns.find((item) => {
+            console.log(`comparing ${headerKey} and ${item.keyName}`);
+            return item.keyName === headerKey;
+        });
+        console.dir(foundWord);
```
Comment (andir) [inline herald/src/app/shared/data-table/data-table.component.ts:40]: how are we gonna deal with those debug statements in the future? Just carry them along? Is there some kind of angular debugging that only works once enabled/in dev mode?

Comment (OliverRadini) [reply herald/src/app/shared/data-table/data-table.component.ts:40]: Would be good, at the moment I'm just logging stuff but it'll need to be changed at some point

**Pull request merged**